kernel/comps/time/src/lib.rs

```diff
@@ -1,39 +1,41 @@
 // SPDX-License-Identifier: MPL-2.0
 
 //! The system time of Asterinas.
+#![feature(let_chains)]
 #![no_std]
 #![deny(unsafe_code)]
 
 extern crate alloc;
 
 use alloc::sync::Arc;
-use core::{sync::atomic::Ordering::Relaxed, time::Duration};
+use core::time::Duration;
 
 use clocksource::ClockSource;
 pub use clocksource::Instant;
 use component::{init_component, ComponentInitError};
+use drivers::{Driver, Rtc};
 use ostd::sync::Mutex;
-use rtc::{get_cmos, is_updating, CENTURY_REGISTER};
 use spin::Once;
 
 mod clocksource;
-mod rtc;
+mod drivers;
 mod tsc;
 
 pub const NANOS_PER_SECOND: u32 = 1_000_000_000;
 pub static VDSO_DATA_HIGH_RES_UPDATE_FN: Once<Arc<dyn Fn(Instant, u64) + Sync + Send>> =
     Once::new();
+static RTC_DRIVER: Once<Rtc> = Once::new();
```

Comment:
Can we declare `RTC_DRIVER` as `Once<Arc<dyn coms::Driver + Send + Sync>>`, thus we do not have to use a macro to implement a enum type `RTC`. In addition, the initialization actions can put in a `init` function in the drivers module.